cluster-api-gcp/CONTRIBUTING.md

```diff
@@ -1,15 +1,54 @@
 # Contributing Guidelines
 
+## Google Cloud Project
+
+If you don't have a Google Cloud Project, please [create one](https://cloud.google.com/resource-manager/docs/creating-managing-projects).
+
+## Set GCP Credentials
+
+In order to use the GCP machine controller, you need to configure the credentials so that the code has access to the GCP project where resources will be created.
+
+Steps to follow:
+1. Verify that the environment variable `GOOGLE_APPLICATION_CREDENTIALS` is set pointing to valid service account credentials
+2. If not set, follow the [instructions on Google Cloud Platform site](https://cloud.google.com/docs/authentication/getting-started) to have it set up.
+
+## Install Google Cloud SDK (gcloud)
+
+Google Cloud SDK (gcloud) will be helpful for two reasons:
+-  Inspect GCP resources during development;
+-  Set configuration values that will be used during development (like project name).
+
+Steps to follow:
+1.  Install as per [Cloud SDK instructions](https://cloud.google.com/sdk/);
+2.  Configure Cloud SDK to point to the GCP project you will be using.
+
+    ```bash
+    $ gcloud auth login
+    $ gcloud config set project <GCP_PROJECT_ID>
+    ```
+
+## Install Docker
+
+1. Install [Docker](https://docs.docker.com/install/) on your machine;
+2. Make sure your user can execute docker commmands (without sudo). This is a way to test it:
+```bash
+$ docker run hello-world
+
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+...
+```
+
 ## Build
 
 ```bash
 $ cd $GOPATH/src/k8s.io/
-$ git clone git@github.com:kubernetes/kube-deploy.git
+$ git clone https://github.com/<GITHUB_USERNAME>/kube-deploy.git
```

Comment:
If you want to use this, you need to have the contributor fork the repo first.

Reply:
Added more details on that.